src/lib/storage/boot_requirements_checker.rb

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require_relative "./proposal_volume"
+require_relative "./disk_size"
+require "pp"
+
+module Yast
+  module Storage
+    #
+    # Class that can check requirements for the different kinds of boot
+    # partition: /boot, EFI-boot, PReP.
+    #
+    # TO DO: Check with arch maintainers if the requirements are correct.
+    #
+    # See also
+    # https://github.com/yast/yast-bootloader/blob/master/SUPPORTED_SCENARIOS.md
+    #
+    class BootRequirementsChecker
+      include Yast::Logger
```

Comment:
Yes. It also doesn't do much yet. So far, it's just very primitive stubs.

I fear that code will grow very quickly, and then there will be lots and lots of logging. ;-)